### ISSUE TITLE
Null reference exception unhandled when source is neither url or html

### ISSUE THREAD

**Issue by gconnolly** [lapwinglabs/x-ray#151]
### Null reference exception unhandled when source is neither url nor html

When crawling the web for urls occasionally my app comes across a malformed url that is fed into x-ray. This causes x-ray to throw an exception due to an null reference.

I understand that this is a "garbage in/garbage out" situation. I can check that the url is well-formed before passing it to x-ray, but it would be nice if x-ray could handle the error properly.

### Your environment

* version of node: v4.2.6
* version of npm: 3.7.3

### Steps to reproduce

```javascript
const Xray = require('x-ray')
const x = Xray()

x('garbageIn', 'title')((err, title) => {
  console.log(title)
})
```

### Expected behaviour

return an error to the callback handler, or fail silently and return undefined

### Actual behaviour

```
TypeError: Cannot read property 'html' of null
    at load (/x-ray-test/node_modules/x-ray/index.js:168:19)
    at node (/x-ray-test/node_modules/x-ray/index.js:108:17)
    at Object.<anonymous> (/x-ray-test/index.js:4:24)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
    at node.js:966:3
```

**Comment by gconnolly** [comment]
I've looked a little into this and will offer my own pull request as a resolution, for what it is worth.

**Comment by gconnolly** [comment]
There are probably a couple of ways to address this issue. Here is my suggestions: #152

**Comment by gconnolly** [comment]
Resolved with #152
